tests/integration/index/query_with_relation_filter_test.go

```diff
@@ -553,3 +553,113 @@ func TestQueryWithIndexOnOneToOne_IfFilterOnIndexedRelation_ShouldFilter(t *test
 
 	testUtils.ExecuteTestCase(t, test)
 }
+
+func TestQueryWithIndexOnManyToOne_IfFilterOnIndexedField_ShouldFilterWithExplain(t *testing.T) {
```

Comment:
question: Was there a behaviour change here?  Before the production code changed what would the results have been?

Reply:
no, there was no change here. The test was passed in the beginning, but I realized that an explicit test was missing for this scenario.

Reply:
What has changed RE publicly visible production behaviour?

Reply:
My second question was not targeted at this particular test, I should have made that clearer.

You have answered that question for this test, and the existing explain one, but what about `TestQueryWithIndexOnManyToOne_IfFilterOnIndexedRelation_ShouldFilterWithExplain` - does that behave the same way before and after the production code change?

Reply:
no, it doesn't. I had to change production code to make it pass

Reply:
What were the results before the production code changed?

Reply:
it returned only 1 related secondary object.

Reply:
You mean primary object? A `Device`?

Reply:
primary is `User`, `Device` is secondary.

Reply:
It is not, device references user, not the other way around.  Not really important though :)

Reply:
it's not important indeed, but I'd like to get the terminology straight.
We usually apply these terms to relations 'User.devices' or 'Device.owner'.
In these examples `User.devices` is secondary and `Device.owner` is primary, right? (I hope I'm not mixing it up)
For me it means is a relation is primary, it refers to primary document, as opposed to being a field of a primary document.

Reply:
> In these examples User.devices is secondary and Device.owner is primary

Yes :) The side that contains the id of the other document is the primary side - it is perhaps easier to see in one-ones.

> For me it means is a relation is primary, it refers to primary document, as opposed to being a field of a primary document.

Ah I see how you are seeing it, that makes sense to me.  We are just talking about different things :)  The primary side of the relation is on `Device`, but the primary object could be seen as the `User`.

Reply:
That means we agree that a self-sufficient object (meaning it has no references to another one) is considered primary. In our case `User`.
And the object that has a reference (as primary relation field) to another object is considered secondary. In this case `Device`.